provision.sh

```diff
@@ -220,8 +229,11 @@ install_aws_cli() {
 
 install_hashicorp_tools() {
   echo ">>> Installing Vault and Terraform"
-  wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor >/usr/share/keyrings/hashicorp-archive-keyring.gpg
-  echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" >/etc/apt/sources.list.d/hashicorp.list
+  rm -f /etc/apt/trusted.gpg.d/hashicorp-archive-keyring.gpg
```

Comment:
If we preserve the redirect from the old version: `>/usr/share/keyrings/hashicorp-archive-keyring.gpg` instead of using `tee` we would not have to add this extra `rm` as the files will be overwritten.